tests/jballerina-debugger-integration-test/src/test/java/org/ballerinalang/debugger/test/adapter/variables/VariableVisibilityTest.java

```diff
@@ -50,36 +52,140 @@ public void setup() throws BallerinaTestException {
+        localVariables = debugTestRunner.fetchVariables(debugHitInfo.getRight(), DebugTestRunner.VariableScope.LOCAL);
+        Assert.assertEquals(localVariables.size(), 36);
+
+        // local variable visibility test inside `foreach` loop.
+        debugTestRunner.resumeProgram(debugHitInfo.getRight(), DebugTestRunner.DebugResumeKind.NEXT_BREAKPOINT);
+        debugHitInfo = debugTestRunner.waitForDebugHit(10000);
+        // TODO - Enable test
+        // Assert.assertEquals(localVariables.size(), 37);
+
+        // local variable visibility test inside `match` statement.
+        debugTestRunner.resumeProgram(debugHitInfo.getRight(), DebugTestRunner.DebugResumeKind.NEXT_BREAKPOINT);
+        debugHitInfo = debugTestRunner.waitForDebugHit(10000);
+        localVariables = debugTestRunner.fetchVariables(debugHitInfo.getRight(), DebugTestRunner.VariableScope.LOCAL);
+        Assert.assertEquals(localVariables.size(), 37);
     }
 
-    @Test
-    public void globalVariableVisibilityTest() {
+    @Test(dependsOnMethods = "controlFlowVariableVisibilityTest",
+        description = "Variable visibility test for global variables")
+    public void globalVariableVisibilityTest() throws BallerinaTestException {
+
+        // global variable visibility test outside main() method.
+        debugTestRunner.resumeProgram(debugHitInfo.getRight(), DebugTestRunner.DebugResumeKind.NEXT_BREAKPOINT);
+        debugHitInfo = debugTestRunner.waitForDebugHit(10000);
+        globalVariables = debugTestRunner.fetchVariables(debugHitInfo.getRight(), VariableScope.GLOBAL);
+        Assert.assertEquals(globalVariables.size(), 14);
+
+        // global variable visibility at the last line of the main() method.
+        debugTestRunner.resumeProgram(debugHitInfo.getRight(), DebugTestRunner.DebugResumeKind.NEXT_BREAKPOINT);
+        debugHitInfo = debugTestRunner.waitForDebugHit(10000);
+        globalVariables = debugTestRunner.fetchVariables(debugHitInfo.getRight(), VariableScope.GLOBAL);
+        localVariables = debugTestRunner.fetchVariables(debugHitInfo.getRight(), VariableScope.LOCAL);
+        Assert.assertEquals(globalVariables.size(), 14);
+
         // Todo - Enable after fixing https://github.com/ballerina-platform/ballerina-lang/issues/26139
-        // Asserts the number of globally visible variables (should be 11).
-        // Assert.assertEquals(globalVariables.size(), 11);
         // global constants
         // debugTestRunner.assertVariable(globalVariables, "nameWithoutType", "Ballerina", "string");
         // debugTestRunner.assertVariable(globalVariables, "nameWithType", "Ballerina", "string");
-
         debugTestRunner.assertVariable(globalVariables, "nameMap", "map", "map");
         debugTestRunner.assertVariable(globalVariables, "nilWithoutType", "()", "nil");
         debugTestRunner.assertVariable(globalVariables, "nilWithType", "()", "nil");
+        debugTestRunner.assertVariable(globalVariables, "RED", "()", "nil");
```

Comment:
This assertion is wrong. The expected value in here should be "RED"

Reply:
Fixed with [7a49014](https://github.com/ballerina-platform/ballerina-lang/pull/28809/commits/7a490140dba6ff648098b8ef1f3162245729dfed)